starterAIs/starter.php

```diff
@@ -0,0 +1,162 @@
+            $ans = "COMPLETE " . $this->index;
+        } else {
+            throw new \Exception("Invalid action");
+        }
+        if ($this->msg != "") {
+            $ans .= " " . $this->msg;
+        }
+        return $ans;
+    }
+}
+
+class Game
+{
+    public $board = null;           // Board
+    public $day = 0;
+    public $nutrients = 0;
+    public $mySun = 0;
+    public $myScore = 0;
+    public $oppSun = 0;
+    public $oppScore = 0;
+    public $oppIsWating = false;
+    public $numberOfTrees = 0;
+    public $trees = [];             // array[int idxCell] of Tree
+    public $possibleMoves = [];     // array of string
+    public $action = null;          // Action
+    public const MAX_DAY = 24;
+    public const EXTRA_POINT_PER_SUNS = 3;
+
+    public function readTurnInput(): void
+    {
+        // $day: the game lasts 24 days: 0-23
+        fscanf(STDIN, "%d", $this->day);
+        $this->startTime = microtime(true);
+        // $nutrients: the base score you gain from the next COMPLETE action
+        fscanf(STDIN, "%d", $this->nutrients);
+        // $mySun: your sun points
+        // $myScore: your current score
+        fscanf(STDIN, "%d %d", $this->mySun, $this->myScore);
+        // $oppSun: opponent's sun points
+        // $oppScore: opponent's score
+        // $oppIsWaiting: whether your opponent is asleep until the next day
+        fscanf(STDIN, "%d %d %d", $this->oppSun, $this->oppScore, $oppIsWaiting);
+        $this->oppIsWating = ($oppIsWaiting == 1);
+        // $numberOfTrees: the current amount of trees
+        fscanf(STDIN, "%d", $this->numberOfTrees);
+        $this->trees = [];
+        for ($i = 0; $i < $this->numberOfTrees; $i++) {
+            // $cellIndex: location of this tree
+            // $size: size of this tree: 0-3
+            // $isMine: 1 if this is your tree
+            // $isDormant: 1 if this tree is dormant
+            fscanf(STDIN, "%d %d %d %d", $cellIndex, $size, $isMine, $isDormant);
+            $tree = new Tree();
+            $tree->cellIndex = $cellIndex;
+            $tree->size = $size;
+            $tree->isMine = ($isMine == 1);
+            $tree->isDormant = ($isDormant == 1);
+            $this->trees[$cellIndex] = $tree;
+        }
+        fscanf(STDIN, "%d", $numberOfPossibleMoves);
+        $this->possibleMoves = [];
+        for ($i = 0; $i < $numberOfPossibleMoves; $i++) {
+            $this->possibleMoves[] = stream_get_line(STDIN, 31 + 1, "\n");
+        }
+    }
+
+    public function output(): void
+    {
+        echo $this->action->toString(), PHP_EOL;
+    }
+
+    public function move(): void
+    {
+        $this->action = new Action(); // defaults to WAIT
+        // TODO
+    }
+}
+
+// --- MAIN ---
+$g = new Game();
+$g->board = new Board();
+$g->board->readInitInput();
+while (true) {
+    $g->readTurnInput();
+    $g->move();
+    $g->output();
+}
+// To debug: error_log(var_export($var, true)); (equivalent to var_dump)
```

Comment:
Maybe you should add a class and function for the error log... is missing the instructions here

Reply:
I kept this commented out "error_log" line from the default solo stubs. Other starter codes do not contain detailed log facilities. Anyhow, I added a debug logState() function to class State and a debug toString() to class Tree